resource/graph_node_test.go

```diff
@@ -132,25 +132,25 @@ func lifecycleTest(t *testing.T, node *resource.GraphNode, initialDeps []string)
 		state, transitionedAt = toState, toTransitionedAt
 	}
 
+	ourErr := errors.New("whoops")
+	node.LogAndSetLastError(ourErr)
+	_, err := node.Resource()
+	test.That(t, err, test.ShouldNotBeNil)
+	test.That(t, err.Error(), test.ShouldContainSubstring, "whoops")
+
+	verifyStateTransition(t, node, resource.NodeStateUnhealthy)
+
 	// mark it for removal
 	test.That(t, node.MarkedForRemoval(), test.ShouldBeFalse)
 	node.MarkForRemoval()
 	test.That(t, node.MarkedForRemoval(), test.ShouldBeTrue)
 
-	_, err := node.Resource()
+	_, err = node.Resource()
 	test.That(t, err, test.ShouldNotBeNil)
 	test.That(t, err.Error(), test.ShouldContainSubstring, "pending removal")
 
 	verifyStateTransition(t, node, resource.NodeStateRemoving)
 
-	ourErr := errors.New("whoops")
-	node.LogAndSetLastError(ourErr)
-	_, err = node.Resource()
-	test.That(t, err, test.ShouldNotBeNil)
-	test.That(t, err.Error(), test.ShouldContainSubstring, "whoops")
-
-	verifyStateTransition(t, node, resource.NodeStateUnhealthy)
-
```

Comment:
can keep here/make a copy as a test for state not transitioning

Reply:
made a copy here to ensure removing -> unhealthy is blocked